front/src/modules/timesStops/helpers/computeMargins.tsx

```diff
@@ -0,0 +1,94 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+import { mToMm, msToS } from 'utils/physics';
+
+import { formatDigitsAndUnit } from './utils';
+
+const emptyMargins = {
+  theoreticalMargin: '',
+  theoreticalMarginSeconds: '',
+  calculatedMarginSeconds: '',
+  diffMarginsSeconds: '',
+};
+
+function getTheoreticalMargin(selectedTrainSchedule: TrainScheduleResult, pathStepId: string) {
+  const pathStepIndex = selectedTrainSchedule.path.findIndex(
+    (pathStep) => pathStep.id === pathStepId
+  );
+  let theoreticalMarginIndex;
+  if (pathStepIndex === 0) {
+    theoreticalMarginIndex = 0;
+  } else {
+    const theoriticalMarginBoundaryIndex = selectedTrainSchedule.margins?.boundaries?.findIndex(
+      (id) => id === pathStepId
+    );
+    if (
+      theoriticalMarginBoundaryIndex === undefined ||
+      theoriticalMarginBoundaryIndex < 0 ||
+      theoriticalMarginBoundaryIndex > selectedTrainSchedule.margins!.values.length - 2
+    ) {
+      return undefined;
+    }
+    theoreticalMarginIndex = theoriticalMarginBoundaryIndex + 1;
+  }
+  return selectedTrainSchedule.margins!.values[theoreticalMarginIndex];
```

Comment:
Assuming the check `... > length - 2` is there to avoid the out of bounds when getting `[index + 1`], the condition is too restrictive by 1.

You can ignore this comment if we do actually want to return undefined on the last value.

Reply:
`boundaries` length is N, `values` length is N + 1.

`boundaryIndex + 1` is out of bounds in `values` if `boundaryIndex + 1 > values.length - 1`

so the condition `boundariesIndex> values.length - 2` is looking fine to me. 

am I missing something ?

Reply:
I got confused on the boundaries vs values, all good